packages/headless/src/features/commerce/instant-products/instant-products-slice.ts

```diff
@@ -69,28 +73,23 @@ export const instantProductsReducer = createReducer(
         }
         const products = cache.products;
 
-        const currentParentIndex = products.findIndex(
-          (product) => product.permanentid === action.payload.parentPermanentId
-        );
+        let childToPromote;
+        const currentParentIndex = products.findIndex((product) => {
```

Comment:
My concern with this approach would be that in cases where lots of products are being displayed (which I think can happen in commerce as the user "browses" a list with the show more button), this could get slower and slower.

I do like the simplicity of this approach, however.

I think we can go ahead with this for now, and keep this on our radar as a potential spot for performance optimization.